### PR TITLE
Use engagementRate instead of engagedSessions in DashboardOverallPageMetricsWidgetGA4.

### PR DESCRIPTION
**Pull request by techanvil**
## Summary

- #6218 

## Relevant technical choices

As discussed on Slack we need to change from using the `engagedSessions` to the `engagementRate` metric.

## PR Author Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 5.2 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [ ] Run the code.
- [ ] Ensure the acceptance criteria are satisfied.
- [ ] Reassess the implementation with the IB.
- [ ] Ensure no unrelated changes are included.
- [ ] Ensure CI checks pass.
- [ ] Check Storybook where applicable.
- [ ] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [ ] Ensure the PR has the correct target branch.
- [ ] Double-check that the PR is okay to be merged.
- [ ] Ensure the corresponding issue has a ZenHub release assigned.
- [ ] Add a changelog message to the issue.

### CODE REVIEW

File: assets/js/modules/analytics-4/utils/data-mock.js [hunk 1]
```diff
@@ -43,6 +43,7 @@ const ANALYTICS_4_METRIC_TYPES = {
 	conversions: 'TYPE_INTEGER',
 	screenPageViews: 'TYPE_INTEGER',
 	engagedSessions: 'TYPE_INTEGER',
+	engagementRate: 'TYPE_INTEGER',
```
Comment (aaemnnosttv) [inline assets/js/modules/analytics-4/utils/data-mock.js:46]: The type here is a float, see real example:

image

Looks like we'd need to add handling for this in `generateMetricValues` as well.

```suggestion
	engagementRate: 'TYPE_FLOAT',
```

**Review by aaemnnosttv — CHANGES_REQUESTED** [review]
Thanks @techanvil, just one thing to address.

File: assets/js/modules/analytics-4/utils/data-mock.js (hunk 1 above)
Comment (techanvil) [reply assets/js/modules/analytics-4/utils/data-mock.js]: Thanks for spotting that! I've updated the widget, and the mock data generator accordingly.

**Review by aaemnnosttv — APPROVED** [review]
LGTM, thanks!

**Pull request merged**